analyzer/src/test/scala/com/avsystem/commons/analyzer/ImplicitValueClassesTest.scala

```diff
@@ -0,0 +1,230 @@
+package com.avsystem.commons
+package analyzer
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class ImplicitValueClassesSuite extends AnyFunSuite with AnalyzerTest {
+  test("implicit final class extending AnyVal should pass") {
+    assertNoErrors(
+      //language=Scala
+      """
+        |object whatever {
+        |  implicit final class GoodImplicitClass(val x: Int) extends AnyVal {
```

Comment:
NIT:

not a fan of implicit classes which have a `val` on their wrapped type - it should be `private val` basically always, otherwise you can do something like this:

```scala
42.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x
```

consider integrating it into this rule (since AnyVals need `val`, but it can be a `private val`)